File: day2/phonebook.cpp
// Example of searching a map

#include <map>
#include <string>
#include <iostream>

using namespace std;


int main()
{
  map<string,string> phonebook;

  while (true) {
    cout << "\nName? ";
    string name;
    getline(cin, name);
    if (name.length() == 0)
      break;

    if (phonebook.find(name) == phonebook.end()) {
      cout << name << " not found!\n";
      cout << "Enter a number for " << name << ": ";
      string number;
      getline(cin, number);
      phonebook[name] = number;
    }
    else {
      cout << name << "'s number is " << phonebook[name] << endl;
    }
  }

  return 0;
}

File: day2/phonebook.py
# python implementation of phonebook.cpp

phonebook = {}

while True:
    name = raw_input("\nName? ")
    if len(name) == 0:
        break

    if name not in phonebook:
        print("not found!")
        enter_number = "Enter a number for {name}: ".format(name=name)
        number = raw_input(enter_number)
        phonebook[name] = number

    else:
        entry = "{name}'s number is {number}".format(name=name,
                                                     number=phonebook[name])
        print(entry)
